src/themes/material/components/edit-chip-list/edit-chip/edit-chip.component.ts

```diff
@@ -0,0 +1,215 @@
+import {
+	Component,
+	Input,
+	Output,
+	ViewChild,
+	EventEmitter,
+	Renderer2,
+	ElementRef
+} from '@angular/core';
+
+import {AppError} from 'ng2-qgrid/core/infrastructure';
+import {Command, CommandManager} from 'ng2-qgrid/core/command';
+import {Shortcut, ShortcutManager} from 'ng2-qgrid/core/shortcut';
+import {DomBase} from 'ng2-qgrid/common/dom-helpers/dom-base';
+
+export const ChipState = {
+	edited: 'edited',
+	new: 'new'
+};
+
+enum NextAction {
+	add,
+	beginEdit,
+	endEdit,
+	nothing
+}
+
+@Component({
+	selector: 'edit-chip',
+	templateUrl: './edit-chip.tpl.html',
+	styleUrls: ['./edit-chip.scss']
+})
+export class EditChipComponent extends DomBase {
+	@Input() index: number;
+	@Input() value: string | any;
+	@Input() state: string;
+	@Input('state') chipState: string;
+
+	@Input('q-grid-focus-disabled')
+	qGridFocusDisabled: boolean = true;
+
+	@Output() delete = new EventEmitter<number>();
+	@Output() add = new EventEmitter<string|any>();
+	@Output() save = new EventEmitter<string|any>();
+
+	@Output('focus-previous') focusPrevious = new EventEmitter<number>();
+	@Output('focus-next') focusNext = new EventEmitter<number>();
+
+	private shortcut = new Shortcut(new ShortcutManager());
+	private nextAction = NextAction.beginEdit;
+
+	private newValue: string;
+	private editValue: string;
+
+	isSelected: boolean = false;
+
+	get isEdited() {
+		return this.chipState === ChipState.edited;
+	}
+
+	private get isNew() {
+		return this.chipState === ChipState.new;
+	}
+
+	private get isReadOnly() {
+		return !(this.isEdited || this.isNew);
+	}
+
+	private get canMoveFocus() {
+		return this.isReadOnly && this.isSelected || this.isNew && !this.newValue;
+	}
+
+	private resetChipState() {
+		this.chipState = '';
+	}
+
+	private beginEditCommand = new Command({
+		canExecute: () => this.nextAction === NextAction.beginEdit && this.isSelected,
+		execute: () => {
+			this.editValue = this.value;
+			this.chipState = ChipState.edited;
+			this.nextAction = NextAction.endEdit;
+		},
+		shortcut: 'enter'
+	});
+
+	private endEditCommand = new Command({
+		canExecute: () => this.nextAction === NextAction.endEdit && this.editValue,
+		execute: () => {
+			this.save.emit(this.editValue);
+			this.endEdit();
+			this.editValue = '';
+		},
+		shortcut: 'enter'
+	});
+
+	private addItemCommand = new Command({
```

Comment:
avoid Command prefix

Reply:
Do you mean 'Command' **postfix**?

Reply:
just think about different names - no suffix

Reply:
Refactored this in the commit: [Merge from Dev: conflicts resolving](https://github.com/qgrid/ng2/pull/39/commits/da05f15a228a611a5059e3049c9f18402446fa8f).